client/src/components/all-components/NewStudentProgress.js

```diff
@@ -0,0 +1,110 @@
+import ALLButton from "./ALLButton";
+import Snackbar from "@mui/material/Snackbar";
+
+const NewStudentProgress = (props) => {
+  const { group, enrolledStudents, assignedLabs } = props;
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const openSnackbar = () => {
+    setSnackbarOpen(true);
+  };
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+
+  const copyToClipboard = () => {
+    navigator.clipboard.writeText(
+      document.getElementById("groupCode").innerHTML.trim(),
+    );
+    openSnackbar();
+  };
+
+  return (
+    <>
+      <a
+        onClick={toggleModal}
+        className="tw-absolute tw-right-0 tw-top-[45%] tw-cursor-pointer
+                tw-font-poppins tw-bg-darkGray tw-text-white tw-p-2 tw-font-medium"
+      >
+        Student Progress
+      </a>
+      <BrandedALLModal
+        direction={"column"}
+        isOpen={modalOpen}
+        toggle={toggleModal}
+      >
+        <div className={"tw-px-[2rem] tw-py-[1rem]"}>
+          <div className={"tw-font-poppins"}>
+            <p className={"tw-title-styling-name tw-mb-3"}>
+              View Group as the Instructor
+            </p>
+            <p className={"tw-title-styling-name tw-text-2xl"}>
+              {group.groupName}
+            </p>
+            <div className={"tw-flex tw-flex-row tw-items-center tw-mb-3"}>
+              <p className={"tw-text-sm"}> Group Code: </p>
+              <p id={"groupCode"} className={"tw-mx-3 tw-font-bold tw-text-md"}>
+                {" "}
+                {group.code}
+              </p>
+              <ALLButton
+                className={"tw-h-[3rem] tw-mb-0"}
+                label={"Copy Code"}
+                onClick={copyToClipboard}
+              />
+            </div>
+            <p className={"tw-w-3/4 tw-text-sm"}>
+              Share this group code with your students to get them enrolled into
+              the class. All they have to do to get started is enter the code!
+            </p>
+            <Snackbar
+              open={snackbarOpen}
+              autoHideDuration={5000}
+              message="Code successfully copied to clipboard!"
+              onClose={() => setSnackbarOpen(false)}
+              anchorOrigin={{ vertical: "top", horizontal: "left" }}
+              className={"tw-font-poppins"}
+              sx={{
+                "& .MuiSnackbarContent-root": {
+                  backgroundColor: "#76b352", // your desired color
+                  color: "white",
+                },
+              }}
+            />
+          </div>
+          {assignedLabs.map((lab, key) => {
+            return (
+              <LabRow
+                key={key}
+                lab={lab}
+                studentProgress={true}
+                group={group}
+                enrolledStudents={enrolledStudents}
+              />
+            );
+          })}
+          ;
```

Comment:
Remove this semicolon
![Screenshot 2024-10-25 at 1 16 03 PM](https://github.com/user-attachments/assets/bc779f69-6981-4d78-a748-12d7a42d7abc)